lib/receivers/tcpReceiver.js

```diff
@@ -68,67 +114,26 @@ module.exports = (() => {
         // an already running processing loop will handle the newly arrived task as well
         lineEnd = data.indexOf('\n');
 
-        while (lineEnd >= 0) {
+        while (lineEnd > -1) {
           task = data.slice(0, lineEnd + 1);
           data = data.slice(lineEnd + 1);
 
-          // TODO: do not ^^^ recopy all the data for each task line, index into data instead
-          // skip blank lines, processBL non-blank ones
-
-          if (task) {
+          if (typeof task === 'undefined' || task == null) {
             break;
           }
 
-          lineEnd = data.indexOf('\n');
-        }
-
-        if (task.indexOf('{"healthCheck":1}') >= 0) {
-          log.info('healthcheck!');
-          socket.write(`${JSON.stringify({ status: 'ready' })}\n`);
-          return;
-        }
-
-        log.debug('About to parse task');
-        log.debug(`Task: ${task}`);
-
-        parseTask(task, (parseError, parsedTask) => {
-          log.debug('Task parsing complete');
+          // TODO: do not ^^^ recopy all the data for each task line, index into data instead
+          // skip blank lines, processBL non-blank ones
 
-          if (parseError) {
-            log.error(`Parse error! ${parseError}`);
-            parseError.isError = true;
-            socket.write(JSON.stringify(composeErrorReply('Internal Error', parseError.toString(), parseError)));
-            socket.write('\n');
+          if (task.indexOf('{"healthCheck":1}') >= 0) {
```

Comment:
Bit of a nit here but above you used `> -1` instead of `>= 0`. Should we be consistent?